atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/async/query/GetQuerySpec.java

```diff
@@ -38,67 +36,74 @@
 
     /**
      * Since each query is constructed for one cell we are using an optimisation that we can ask the CQL to do most of
-     * the work internally. First of all we are using the fact that timestamps in {@code column2} are ordered in
-     * {@code ASC} order and since we are interested in the most recent timestamp we use {@code LIMIT 1} to get the
-     * latest value. This should help with both cassandra workload and amount of transferred data.
+     * the work internally. First of all we are using the fact that cells are ordered by {@code column} and
+     * {@code timestamp} value in {@code ASC} order and since we are interested in the value with the most recent
```

Comment:
might the mention of us storing timestamps as negative timestamps make this make a bit more sense? @jeremyk-91

Reply:
Yep. I'm not sure why the `column` (presumably `column1`) ordering matters here since it's a single SELECT query? The important bit is that `column2` in the clustering key is ASC, and it is the timestamps stored in reverse bit order, so we can get the value with the largest real timestamp with a LIMIT 1.

Reply:
Added a clarifying comment about how timestamps are stored.